### DISCUSSION TITLE
Caching?

### DISCUSSION THREAD

**Discussion by TUBEOF** [spotify-web-api-java/spotify-web-api-java#400] [Q&A] [answered]
Does this Wrapper automatically cache requests and re-use the results?
For example, when i call `GetUsersAvailableDevicesRequest` and a few seconds later, again - does it call the Spotify API again or is a cached response used automatically?

If not: Is it somehow possible to do so, like with some HttpClient settings, which i can set?

**Comment by TUBEOF** [comment]
Otherwise, I would need to go with Guava Cache, but just want to make sure, to avoid useless work

**Comment by dargmuesli** [comment, marked as answer]
Yes, [give `cache` a search in this repository](https://github.com/search?q=repo%3Aspotify-web-api-java%2Fspotify-web-api-java%20cache&type=code) and you quickly find [what you were thinking of](https://github.com/spotify-web-api-java/spotify-web-api-java/blob/1566e455bc9e3c04a26a9615cc9c40a36e9f7568/src/main/java/se/michaelthelin/spotify/SpotifyHttpManager.java#L108).
